cmd/dockerd/daemon_windows.go

```diff
@@ -59,7 +59,14 @@ func notifyShutdown(err error) {
 }
 
 func (cli *DaemonCli) getPlatformContainerdDaemonOpts() ([]supervisor.DaemonOpt, error) {
-	return nil, nil
+	opts := []supervisor.DaemonOpt{
+		// On Windows, it first checks if a containerd binary is found in the same
+		// directory as the dockerd binary. If found, this binary takes precedence
+		// over containerd binaries installed in $PATH.
+		supervisor.WithDetectLocalBinary(),
+	}
```

Comment:
I moved the detection to the initialisation as a `DaemonOpt`; this approach made it slightly more flexible (on Windows we use this option, on Linux not (yet?)); performing the lookup during initialisation stores the result in the supervisor's config.

The previous iteration performed the lookup during `startContainerd()`, which is called in `monitorDaemon()`, which re-starts containerd if it would fail, and (theoretically) would mean that deleting the `containerd.exe` binary would cause `dockerd` to lookup the binary again, and switch from a bundled containerd to a system containerd. I think that is undesirable, so it's better to _fail_ in that case than to switch to a different binary.

https://github.com/moby/moby/blob/3a7fda78a28e7a3f161d15d0ae04d9aeaa591209/libcontainerd/supervisor/remote_daemon.go#L298

Reply:
I think we probably _shouldn't_ use this on Linux unless we also disable it for the case of `dockerd` in a common path like `/usr/bin`, `/usr/sbin`, etc. :sweat_smile:

(This seems like a sane approach regardless though, to be clear :+1:)